core/persistence-jpa/src/main/java/org/apache/syncope/core/persistence/jpa/MasterDomain.java

```diff
@@ -149,34 +156,37 @@ public DomainEntityManagerFactoryBean masterEntityManagerFactory() {
         DomainEntityManagerFactoryBean masterEntityManagerFactory = new DomainEntityManagerFactoryBean();
         masterEntityManagerFactory.setMappingResources(orm);
         masterEntityManagerFactory.setPersistenceUnitName("Master");
-        masterEntityManagerFactory.setDataSource((DataSource) masterDataSource().getObject());
+
```

Comment:
I don't get the rationale of this change, e.g. from

```
 masterEntityManagerFactory.setDataSource((DataSource) masterDataSource().getObject());
```

to

```
        DataSource dataSource = (DataSource) masterDataSource().getObject();
        masterEntityManagerFactory.setDataSource(Objects.requireNonNull(dataSource));
```

Why not leaving it as single statement?

Reply:
Static analysis did complain that `(DataSource) masterDataSource().getObject();` can in in fact be null, which would then ultimately cause an NPE down the road. It appears that underlying components do catch null at various levels, but it might be good to prevent it upfront.

Reply:
Agree, but why not simply changing into:

```
 masterEntityManagerFactory.setDataSource(Objects.requireNonNull((DataSource) masterDataSource().getObject()));
```

Reply:
Sure thing.